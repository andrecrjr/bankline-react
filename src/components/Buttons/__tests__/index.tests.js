import React from "react";
import { shallow, mount } from "enzyme";
import { ButtonPay, BackButton } from "../index";
import toJson from "enzyme-to-json";

jest.mock("react-router-dom", () => ({
  useHistory: () => ({
    push: jest.fn(),
    location: { pathname: "/pay" }
  })
}));

describe("It should mount buttons", () => {
  it("It button Pay renders", () => {
    const buttonPay = shallow(<ButtonPay link={"/"}>Pagar</ButtonPay>);

    expect(toJson(buttonPay)).toMatchSnapshot();
  });
  it("It renders back button", () => {
    const buttonPay = shallow(<BackButton>Voltar</BackButton>);
    expect(toJson(buttonPay)).toMatchSnapshot();
  });
  it("It button Pay renders with width and margin", () => {
    const clickPush = jest.fn();
    const buttonPay = shallow(
      <ButtonPay width={"100px"} margin={"50px"}>
        Pagar
      </ButtonPay>
    );
    buttonPay.find("button").simulate("click");
    expect(clickPush).to.equal(1);
  });
});
