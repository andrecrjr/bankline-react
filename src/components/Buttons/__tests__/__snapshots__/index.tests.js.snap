// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should mount buttons It button Pay renders 1`] = `
<Component
  to="/"
>
  <button
    className="main--pay"
    type="submit"
  >
    Pagar
  </button>
</Component>
`;

exports[`It should mount buttons It renders back button 1`] = `
<span
  className="page-user__back-button"
  onClick={[Function]}
>
  <img
    src="arrowBack.svg"
  />
</span>
`;
